### PR TITLE
Added WiFi Manager and changed examples

### PR DESCRIPTION
**Pull request by makermelissa**
I added a WiFi manager to handle a lot of the connection details and change the status light on the board. I also updated Jerry's examples to make use of the WiFi Manager.

### CODE REVIEW

File: examples/esp32spi_cheerlights.py [hunk 1]
```diff
@@ -4,87 +4,53 @@
...
-        for x in DATA_LOCATION:
-            value = value[x]
+        response = wifi.get(DATA_SOURCE)
+        print(response.json())
+        value=response.json()
+        for key in DATA_LOCATION:
+            value = value[key]
             print(value)
-        r.close()
+        response.close()
```
Comment (jerryneedell) [inline examples/esp32spi_cheerlights.py]: With change to response, line 66 should also replace r with response

Comment (makermelissa) [reply examples/esp32spi_cheerlights.py]: Oh, good catch

File: examples/esp32spi_cheerlights.py [hunk 2]
```diff
@@ -4,87 +4,53 @@
...
 # neopixels
 pixels = neopixel.NeoPixel(board.A1, 16, brightness=0.3)
 pixels.fill(0)
-builtin = neopixel.NeoPixel(board.NEOPIXEL, 1, brightness=0.1)
-builtin[0] = 0
 
 # we'll save the value in question
 last_value = value = None
 the_time = None
 times = 0
```
Comment (jerryneedell) [inline examples/esp32spi_cheerlights.py]: Not your change, but I don’t think times or the_time are used. Remove here and the increment of times below.

**Comment by jerryneedell** [comment]
This looks great. I’d prefer to be able to test it before approval. I should be able to do that tomorrow morning.

**Comment by makermelissa** [comment]
Ok done. I also added the examples to the RTD examples file.

**Comment by makermelissa** [comment]
I decided to add params to the DocStrings just to polish it up a little more.

File: adafruit_esp32spi/adafruit_esp32spi_wifimanager.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            try:
+                if self.debug:
+                    print("Connecting to AP...")
+                self.neo_status((100, 0, 0))
+                self._esp.connect_AP(bytes(self.ssid, 'utf-8'), bytes(self.password, 'utf-8'))
+                self.neo_status((0, 100, 0))
+            except (ValueError, RuntimeError) as error:
+                if self.debug:
+                    print("Failed to connect, retrying\n", error)
+                continue
```
Comment (jerryneedell) [inline adafruit_esp32spi/adafruit_esp32spi_wifimanager.py]: should this failure message be printed even if debug not enabled? As it is, it just fails silently and keeps retrying.

**Comment by jerryneedell** [comment]
tested post and cheerlights examples on metro_m4_espress with Huzzah ESP32 and feather_nrt52840_express with Espressif DevKit  -- both worked great!

File: adafruit_esp32spi/adafruit_esp32spi_wifimanager.py (hunk 3 above)
Comment (makermelissa) [reply adafruit_esp32spi/adafruit_esp32spi_wifimanager.py]: Yeah, I can remove the only show on debug condition

**Comment by ladyada** [comment]
@kattni wanna bump this too?

**Comment by kattni** [comment]
@ladyada @makermelissa has been keeping up with bumping libs like a champ! But I can take care of this one to get it done!

**Comment by makermelissa** [comment]
Is bumping the same thing as pushing a release in this case?

**Comment by ladyada** [comment]
ooh pardon - ok flip a coin :) yes when i say 'bump' i mean you increase the semver and release

**Comment by kattni** [comment]
@makermelissa Yep! That's exactly what it is :)

**Comment by kattni** [comment]
Already released!

**Comment by makermelissa** [comment]
Ah, ok. Thanks.

**Pull request merged**